File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/DefinitionList.kt
package com.nouprax.markdown.core

public class DefinitionList internal constructor(
    public val definitions: kotlin.collections.List<Definition>,
    override val scope: Scope,
    override val anchor: String?,
    override val attributes: Attributes,
) : Markup {
    override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visitDefinitionList(this)
}

File: packages/es-markdown-core/src/markup/definition-list.ts
import type { MarkupBase } from "./base.js";
import type { Markup } from "./markup.js";

export interface DefinitionList extends MarkupBase<"definitionList"> {
    readonly definitions: readonly Definition[];
}

export interface Definition extends MarkupBase<"definition"> {
    readonly term: readonly Markup[];
    readonly content: readonly (readonly Markup[])[];
    readonly compact: boolean;
}
